contracts/middle-layer/TokenHubV2.sol

```diff
@@ -0,0 +1,298 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
```

Comment:
I expected more gas profile testing on the resource mirror, especially the grouphub.

The current implementation is not complete.